### PR TITLE
Auth: Move star service to identity interface

### PR DESCRIPTION
**Pull request by Jguer**
**What is this feature?**

- Use identity interface for star service
- Blocks users without a user ID from fetching stars (they pass ID=0 to store)
- Add basic test structure for star service API

**Why do we need this feature?**

[Add a description of the problem the feature is trying to solve.]

**Who is this feature for?**

[Add information on what kind of user the feature is for.]

**Which issue(s) does this PR fix?**:

Fixes #

**Special notes for your reviewer:**

Please check that:
- [x] It works as expected from a user's perspective.
- [x] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: pkg/services/auth/identity/requester.go [hunk 1]
```diff
@@ -50,3 +59,24 @@ type Requester interface {
 	// HasUniqueId returns true if the entity has a unique id
 	HasUniqueId() bool
 }
+
+// IntIdentifier converts a string identifier to an int64.
+// Applicable for users, service accounts and api keys.
```
Comment (IevaVasiljeva) [inline pkg/services/auth/identity/requester.go:64]:
```suggestion
// Applicable for users, service accounts, api keys and renderer service.
```

File: pkg/services/star/api/api.go [hunk 2]
```diff
@@ -39,15 +40,25 @@ func (api *API) getDashboardHelper(ctx context.Context, orgID int64, id int64, u
...
+		return nil, response.Error(http.StatusNotFound, "Dashboard not found", err)
 	}
 
 	return result, nil
 }
 
 func (api *API) GetStars(c *contextmodel.ReqContext) response.Response {
+	namespace, identifier := c.SignedInUser.GetNamespacedID()
+	if namespace != identity.NamespaceUser && namespace != identity.NamespaceServiceAccount {
+		return response.Error(http.StatusBadRequest, "Only users and service accounts can star dashboards", nil)
```
Comment (IevaVasiljeva) [inline pkg/services/star/api/api.go:52]: It's actually pretty weird that service accounts can star dashboards. But oh well...

**Review by IevaVasiljeva — APPROVED** [review]
LGTM!

File: pkg/services/star/api/api.go (hunk 2 above)
Comment (Jguer) [reply pkg/services/star/api/api.go:52]: Didn't want to break what existed :sweat_smile: but yeah.

File: pkg/services/contexthandler/model/model.go [hunk 3]
```diff
@@ -20,6 +20,7 @@ import (
 type ReqContext struct {
 	*web.Context
 	*user.SignedInUser
+	// SignedInUser identity.Requester
```
[the remark this answers is not in the document]
Comment (Jguer) [reply pkg/services/contexthandler/model/model.go:23]: that's an artifact, I'll fix it :)

Comment (Jguer) [inline pkg/services/contexthandler/model/model.go:23]:
```suggestion
```

**Pull request merged**